assets/training/distillation/components/data_generation_batch_scoring_postprocess/spec.yaml

```diff
@@ -0,0 +1,105 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
+name: oss_distillation_generate_data_batch_postprocess
+version: 0.0.1
+type: command
+
+is_deterministic: True
+
+display_name: OSS Distillation Generate Data Postprocess Batch Scoring
+description: Component to prepare data returned from teacher model enpoint in batch 
+
+environment: azureml://registries/azureml/environments/acft-hf-nlp-gpu/labels/latest
+
+inputs:
+  # Inputs
+  train_file_path:
+    type: uri_file
+    description: Path to the registered training data asset. The supported data formats are `jsonl`, `json`, `csv`, `tsv` and `parquet`.
+    mode: rw_mount
+
+  validation_file_path:
+    type: uri_file
+    optional: true
+    description: Path to the registered validation data asset. The supported data formats are `jsonl`, `json`, `csv`, `tsv` and `parquet`.
+    mode: rw_mount
+
+  hash_train_data:
+    type: uri_file
+    optional: false
+    description: jsonl file containing the hash for each payload.
+
+  hash_validation_data:
+    type: uri_file
+    optional: true
+    description: jsonl file containing the hash for each payload.
+
+  batch_score_train_result: 
+    type: uri_folder
+    description: Path to the directory containing jsonl file(s) that have the result for each payload.
+
+  batch_score_validation_result: 
+    type: uri_folder
+    optional: true
+    description: Path to the directory containing jsonl file(s) that have the result for each payload.
+
+  min_endpoint_success_ratio:
+    type: number
+    default: 0.7
+    description: >
+      The minimum value of (successful_requests / total_requests) required for classifying inference as successful.
+      If (successful_requests / total_requests) < min_endpoint_success_ratio, the experiment will be marked as failed. 
+      By default it is 0.7 (0 means all requests are allowed to fail while 1 means no request should fail.)
+
+  enable_chain_of_thought:
+    type: string
+    optional: true
+    default: "false"
+    description: Enable Chain of thought for data generation
+  
+  enable_chain_of_density:
+    type: string
+    optional: true
+    default: "false"
+    description: Enable Chain of density for text summarization
+
+  data_generation_task_type:
+    type: string
+    enum:
+      - NLI
+      - CONVERSATION
+      - NLU_QA
+      - MATH
+      - SUMMARIZATION
+    description: >
+      Data generation task type. Supported values are:
+      1. NLI: Generate Natural Language Inference data
+      2. CONVERSATION: Generate conversational data (multi/single turn)
+      3. NLU_QA: Generate Natural Language Understanding data for Question Answering data
+      4. MATH: Generate Math data for numerical responses
+      5. SUMMARIZATION: Generate Key Summary for an Article
+    
+outputs:
+  generated_batch_train_file_path:
+    type: uri_file
+    description: Generated train data
+    mode: rw_mount
+  generated_batch_validation_file_path:
```

Comment:
does it make sense to create two instances of this in the main pipeline, one for training and one for validation data, instead of handling both in a single pipeline? so the input would contain: 

1. input_data
2. hash_data
3. batch_score_result

and output would be

1. output_data

Reply:
@ayush-ms - As discussed in the PR review, closing this conversation

Reply:
@visahan-24 , you have tagged wrong person